UltraStar Play/Assets/Scenes/Sing/PlayerController/PlayerScoreController.cs

```diff
@@ -135,39 +135,44 @@ private void OnBeatAnalyzed(PlayerPitchTracker.BeatAnalyzedEvent beatAnalyzedEve
         }
 
         // The beat was sung correctly.
-        if (!noteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
+        if (!ScoreData.NoteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
         {
             noteScore = new NoteScore(analyzedNote);
-            noteToNoteScoreMap.Add(analyzedNote, noteScore);
+            ScoreData.NoteToNoteScoreMap.Add(analyzedNote, noteScore);
         }
-        noteScore.correctlySungBeats++;
+        noteScore.CorrectlySungBeats++;
 
         Sentence analyzedSentence = beatAnalyzedEvent.NoteAtBeat.Sentence;
-        if (!sentenceToSentenceScoreMap.TryGetValue(analyzedSentence, out SentenceScore sentenceScore))
+        if (!ScoreData.SentenceToSentenceScoreMap.TryGetValue(analyzedSentence, out SentenceScore sentenceScore))
         {
             sentenceScore = new SentenceScore(analyzedSentence);
-            sentenceToSentenceScoreMap.Add(analyzedSentence, sentenceScore);
+            ScoreData.SentenceToSentenceScoreMap.Add(analyzedSentence, sentenceScore);
         }
 
         if (analyzedNote.IsNormal)
         {
-            correctNormalNoteLengthTotal++;
+            ScoreData.CorrectNormalNoteLengthTotal++;
             sentenceScore.CorrectlySungNormalBeats++;
         }
         else if (analyzedNote.IsGolden)
         {
-            correctGoldenNoteLengthTotal++;
+            ScoreData.CorrectGoldenNoteLengthTotal++;
             sentenceScore.CorrectlySungGoldenBeats++;
         }
     }
 
     private void OnNoteAnalyzed(PlayerPitchTracker.NoteAnalyzedEvent noteAnalyzedEvent)
     {
+        if (noteAnalyzedEvent.Note.EndBeat < NextBeatToScore)
+        {
+            return;
+        }
+
         Note analyzedNote = noteAnalyzedEvent.Note;
-        if (noteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
+        if (ScoreData.NoteToNoteScoreMap.TryGetValue(analyzedNote, out NoteScore noteScore))
         {
             //Debug.Log($"OnNoteAnalyzed: {noteScore.correctlySungBeats} / {analyzedNote.Length}, {analyzedNote.StartBeat}, {analyzedNote.EndBeat}, {analyzedNote.Text}");
-            if (noteScore.correctlySungBeats >= analyzedNote.Length)
+            if (noteScore.CorrectlySungBeats >= analyzedNote.Length)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/manual/UltraStar-Deluxe/Play/pullRequest?prid=5520139)